include/secp256k1_musig.h

```diff
@@ -375,26 +171,207 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
  *  encoded numbers are out of range, signature verification with it is
  *  guaranteed to fail for every message and public key.
  */
-SECP256K1_API int secp256k1_musig_partial_signature_parse(
+SECP256K1_API int secp256k1_musig_partial_sig_parse(
     const secp256k1_context* ctx,
-    secp256k1_musig_partial_signature* sig,
+    secp256k1_musig_partial_sig* sig,
     const unsigned char *in32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Computes a aggregate public key and the hash of the given public keys.
+ *
+ *  Different orders of `pubkeys` result in different `agg_pk`s.
+ *
+ *  The pubkeys can be sorted before combining with `secp256k1_xonly_sort` which
+ *  ensures the same resulting `agg_pk` for the same multiset of pubkeys.
+ *  This is useful to do before pubkey_agg, such that the order of pubkeys
+ *  does not affect the aggregate public key.
+ *
+ *  Returns: 1 if the public keys were successfully aggregated, 0 otherwise
+ *  Args:        ctx: pointer to a context object initialized for verification
+ *           scratch: scratch space used to compute the aggregate pubkey by
+ *                    multiexponentiation. If NULL, an inefficient algorithm is used.
+ *  Out:      agg_pk: the MuSig-aggregated xonly public key. If you do not need it,
+ *                    this arg can be NULL.
+ *      keyagg_cache: if non-NULL, pointer to a musig_keyagg_cache struct that
+ *                    is required for signing (or verifying the MuSig protocol).
+ *   In:     pubkeys: input array of pointers to public keys to aggregate. The order
+ *                    is important; a different order will result in a different
+ *                    aggregate public key
+ *         n_pubkeys: length of pubkeys array. Must be greater than 0.
+ */
+SECP256K1_API int secp256k1_musig_pubkey_agg(
+    const secp256k1_context* ctx,
+    secp256k1_scratch_space *scratch,
+    secp256k1_xonly_pubkey *agg_pk,
+    secp256k1_musig_keyagg_cache *keyagg_cache,
+    const secp256k1_xonly_pubkey * const* pubkeys,
+    size_t n_pubkeys
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+
+/** Tweak an x-only public key by adding the generator multiplied with tweak32
+ *  to it. The resulting output_pubkey with the original agg_pk output of
+ *  musig_pubkey_agg and tweak passes `secp256k1_xonly_pubkey_tweak_test`.
+ *
+ *  This function is only useful before initializing a signing session. If you
+ *  are only computing a public key, but not intending to create a signature for
+ *  it, you can just use `secp256k1_xonly_pubkey_tweak_add`. Can only be called
+ *  once with a given keyagg_cache.
```

Comment:
This really confuses me,
1. It says it computes an xonly pubkey with `P + tweak32*G`,  but there's no input pubkey, and nothing here is xonly.
2. it talks about `secp256k1_xonly_pubkey_tweak_test` which as far as I could tell doesn't exists?
3. it says that if you just want to compute the pubkey you shouldn't use this but instead use the regular tweak add, but when should you use this? and what exactly does it do?
4. Also the last sentence is bundled together in the same line even though its a separate property (that you shouldn't call this function twice with the same keyagg_cache)

Reply:
So for example if your MuSig key is a taproot internal key and you want to sign for the taproot key?
And I'm guessing that the aggkey is inside of the cache? (so the cache is both a cache and a state)